src/gui_common/charts/DataPoint.cs

```diff
@@ -40,6 +40,9 @@ public enum MarkerIcon
         Skull,
     }
 
+    public double X { get; private set; }
+    public double Y { get; private set; }
```

Comment:
Need to check that hash code computation is still correct, last time I made these readonlies because otherwise DataPoint dictionaries end up failing to fetch the correct entries.

Reply:
As a cache, the x and y won't be set when there are external references. If the x and y is changed but reference is still held in a dict, it is a bug that needs to be fixef.

Reply:
If X and Y are used in GetHashCode, then this is pretty dangerous to have these mutable...
Maybe the design could be made safer by not overriding the hash code getting for this class, and instead in charts have an extra dictionary mapping immutable coordinates to data points? For example `Dictionary<(double X, double Y), DataPoint>`

Reply:
So reading the code it turns out that the cache doesn't care about the properties. But is there a conclusion on whether this had any impact on the chart class? I guess as these are private it is good enough even without a comment as only when getting a point from cache it can be modified.